src/main/java/ch/njol/skript/variables/Variables.java

```diff
@@ -36,241 +61,301 @@
+					else
+						break; // variables loaded, exit thread
 				}
 			}
-		};
+		});
 		loadingLoggerThread.start();
-		
+
 		try {
 			boolean successful = true;
-			for (final Node node : (SectionNode) databases) {
+
+			for (Node node : (SectionNode) databases) {
 				if (node instanceof SectionNode) {
-					final SectionNode n = (SectionNode) node;
-					final String type = n.getValue("type");
+					SectionNode sectionNode = (SectionNode) node;
+
+					String type = sectionNode.getValue("type");
 					if (type == null) {
 						Skript.error("Missing entry 'type' in database definition");
 						successful = false;
 						continue;
 					}
-					
-					final String name = n.getKey();
+
+					String name = sectionNode.getKey();
 					assert name != null;
-					final VariablesStorage s;
-					if (type.equalsIgnoreCase("csv") || type.equalsIgnoreCase("file") || type.equalsIgnoreCase("flatfile")) {
-						s = new FlatFileStorage(name);
+
+					// Initiate the right VariablesStorage class
+					VariablesStorage variablesStorage;
+					if (type.equalsIgnoreCase("csv")
+							|| type.equalsIgnoreCase("file")
+							|| type.equalsIgnoreCase("flatfile")) {
+						variablesStorage = new FlatFileStorage(name);
 					} else if (type.equalsIgnoreCase("mysql")) {
-						s = new DatabaseStorage(name, Type.MYSQL);
+						variablesStorage = new DatabaseStorage(name, Type.MYSQL);
 					} else if (type.equalsIgnoreCase("sqlite")) {
-						s = new DatabaseStorage(name, Type.SQLITE);
+						variablesStorage = new DatabaseStorage(name, Type.SQLITE);
 					} else {
 						if (!type.equalsIgnoreCase("disabled") && !type.equalsIgnoreCase("none")) {
 							Skript.error("Invalid database type '" + type + "'");
 							successful = false;
 						}
+
 						continue;
 					}
-					
-					final int x;
-					synchronized (tempVars) {
-						final Map<String, NonNullPair<Object, VariablesStorage>> tvs = tempVars.get();
+
+					// Get the amount of variables currently loaded
+					int totalVariablesLoaded;
+					synchronized (TEMP_VARIABLES) {
+						Map<String, NonNullPair<Object, VariablesStorage>> tvs = TEMP_VARIABLES.get();
 						assert tvs != null;
-						x = tvs.size();
+						totalVariablesLoaded = tvs.size();
 					}
-					final long start = System.currentTimeMillis();
+
+					long start = System.currentTimeMillis();
 					if (Skript.logVeryHigh())
 						Skript.info("Loading database '" + node.getKey() + "'...");
-					
-					if (s.load(n))
-						storages.add(s);
+
+					// Load the variables
+					if (variablesStorage.load(sectionNode))
+						STORAGES.add(variablesStorage);
 					else
 						successful = false;
-					
-					final int d;
-					synchronized (tempVars) {
-						final Map<String, NonNullPair<Object, VariablesStorage>> tvs = tempVars.get();
+
+					// Get the amount of variables loaded by this variables storage object
+					int newVariablesLoaded;
+					synchronized (TEMP_VARIABLES) {
```

Comment:
Ideally we would switch to something in main memory rather than synchronizing on all of these to save time.

Reply:
What exactly do you mean?
I'm not sure if anything related to the way we synchronize here saves any relevant time, this code is only called once when loading all variables.

Reply:
If that's the only time it's used it's probably fine, though if that's the case does it need to be synchronized at all? When would this be accessed by another process?

Reply:
As far as I can see, it's accessed by 2 threads: the main Bukkit thread, and the logger thread (that reports the status updates, i.e. `Loaded X variables so far...`)